firebase-firestore/src/test/java/com/google/firebase/firestore/remote/BloomFilterTest.java

```diff
@@ -0,0 +1,252 @@
+  }
+
+  @Test
+  public void constructorShouldThrowIAEOnNegativeHashCount() {
+    IllegalArgumentException emptyBloomFilterException =
+        assertThrows(IllegalArgumentException.class, () -> new BloomFilter(new byte[0], 0, -1));
+    assertThat(emptyBloomFilterException).hasMessageThat().contains("Invalid hash count: -1");
+    IllegalArgumentException nonEmptyBloomFilterException =
+        assertThrows(IllegalArgumentException.class, () -> new BloomFilter(new byte[] {1}, 1, -1));
+    assertThat(nonEmptyBloomFilterException).hasMessageThat().contains("Invalid hash count: -1");
+  }
+
+  @Test
+  public void constructorShouldThrowIAEOnOverflowPadding() {
+    IllegalArgumentException exception =
+        assertThrows(IllegalArgumentException.class, () -> new BloomFilter(new byte[] {1}, 8, 1));
+    assertThat(exception).hasMessageThat().contains("Invalid padding: 8");
+  }
+
+  @Test
+  public void mightContainCanProcessNonStandardCharacters() {
+    // A non-empty BloomFilter object with 1 insertion : "ÀÒ∑"
+    BloomFilter bloomFilter = new BloomFilter(new byte[] {(byte) 237, 5}, 5, 8);
+    assertTrue(bloomFilter.mightContain("ÀÒ∑"));
+    assertFalse(bloomFilter.mightContain("Ò∑À"));
+  }
+
+  @Test
+  public void mightContainOnEmptyBloomFilterShouldReturnFalse() {
+    BloomFilter bloomFilter = new BloomFilter(new byte[0], 0, 0);
+    assertFalse(bloomFilter.mightContain("a"));
+  }
+
+  @Test
+  public void mightContainWithEmptyStringShouldReturnFalse() {
+    BloomFilter emptyBloomFilter = new BloomFilter(new byte[0], 0, 0);
+    assertFalse(emptyBloomFilter.mightContain(""));
+    BloomFilter nonEmptyBloomFilter = new BloomFilter(new byte[] {(byte) 255}, 0, 1);
+    assertFalse(nonEmptyBloomFilter.mightContain(""));
+  }
+
+  @Test
+  public void bloomFilterToString() {
+    BloomFilter emptyBloomFilter = new BloomFilter(new byte[0], 0, 0);
+    assertEquals(emptyBloomFilter.toString(), "BloomFilter{hashCount=0, size=0, bitmap=\"\"}");
+    BloomFilter nonEmptyBloomFilter = new BloomFilter(new byte[] {1}, 1, 1);
+    assertEquals(
+        nonEmptyBloomFilter.toString(), "BloomFilter{hashCount=1, size=7, bitmap=\"AQ==\"}");
+  }
+
+  /**
+   * Golden tests are generated by backend based on inserting n number of document paths into a
+   * bloom filter.
+   *
+   * <p>Full document path is generated by concatenating documentPrefix and number n, eg,
+   * projects/project-1/databases/database-1/documents/coll/doc12.
+   *
+   * <p>The test result is generated by checking the membership of documents from documentPrefix+0
+   * to documentPrefix+2n. The membership results from 0 to n is expected to be true, and the
+   * membership results from n to 2n is expected to be false with some false positive results.
+   */
+  private void runGoldenTest(String testFile) throws Exception {
+    String resultFile = testFile.replace("bloom_filter_proto", "membership_test_result");
```

Comment:
Verify that `resultFile.equals(testFile) == false`. If the `testFile` does not contain "bloom_filter_proto" then both `testFile` and `resultFile` will be equal, which is not right. Throw IllegalArgumentException in this case.